### ISSUE TITLE
Can't build demos due to missing s9w_geom_types.h file.

### ISSUE THREAD

**Issue by azawadzki** [s9w/oof#2]
```
~/Development/oof/demos master g++-11 main.cpp 
In file included from main.cpp:3:
tools.h:10:10: fatal error: s9w/s9w_geom_types.h: No such file or directory
   10 | #include <s9w/s9w_geom_types.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Comment by azawadzki** [comment]
Ah, just noticed that the missing library is referenced in the readme:
```
The source code from the demo videos at the beginning is in this repo under [demos/](demos). That code uses a not-included and yet unreleased helper library (`s9w::`) for colors and math. But those aren't crucial if you just want to have a look.
```

**Comment by s9w** [comment]
Hi! Yeah, the situation is this: While the demos may seem simple, and are in most ways - they do require at least three things that devs often need:
1) A fast and generally usable RNG (so certainly not `<random>`)
2) Basic linear algebra
3) Some color code (in this case: non-rgb color spaces and color blending)

You can solve 2) with GLM and 1) with lots of bits of code. 3) is a bit tricky in C++. I solved all of these with my own "s9w" library. It's not mature enough for release yet really, so I didn't include it.

If you want, I could include the version I have. I thought the demo code would be helpful without being able to compile. But I can see this being a less-than-ideal situation.

**Comment by azawadzki** [comment]
> If you want, I could include the version I have. I thought the demo code would be helpful without being able to compile. But I can see this being a less-than-ideal situation.

It would be lovely to be able to compile and run the code on one's system, but if that's not supported yet, then so be it. I'm sure that potential users of a brand new free library can wait a bit (especially, given how thorough the readme is). Congratulations on great presentation, btw!

**Comment by s9w** [comment]
Aww, everyone is so nice today! Thank you.

I zipped the four required s9w headers. I assume you tried this on windows? Setting up the paths so they can be found by Visual Studio or whatever your compiler might be should be straightforward. 

Let me know if this works or if you need help. I hope I didn't forget anything.

~[s9w.zip](https://github.com/s9w/oof/files/7584286/s9w.zip)~

**Comment by azawadzki** [comment]
Looks like I still need `s9w_rng.h`.

**Comment by s9w** [comment]
Yes.. just noticed, sorry. [s9w.zip](https://github.com/s9w/oof/files/7584345/s9w.zip)

Btw I really hope you're on windows. For some window size functionality, mouse cursor etc I use windows-API functions in the demo.

**Comment by azawadzki** [comment]
Thanks! No, I'm actually on macOS. Looks like I'll need to modify the sources a bit to make the demos work in environment, which I'm happy to do, though it will take a while until I report back :) Thanks again for your prompt help!

**Comment by s9w** [comment]
Ah! That sucks for the moment. But there's great opportunity in this: Someone actually testing this on a non-windows machine would be awesome. Maybe try easy things first though. See if the VT sequences actually work. And no idea about how performant macos console is.

From the windows API, I mostly used functions to retrieve the mouse cursor and console window size I think. But the mouse cursor position is only for the color_trail demo. And the window size you can just set something fixed for the moment. Maybe that'll be enough to get things going.

I might also make another issue tomorrow or so about VT query commands, which allow information to be retrieved via VT sequences. But to read them I need OS-dependent functions, so an mac guy would be cool.

Anyways, hit me up if I can be of any help.
